service/src/main/java/bio/terra/workspace/service/workspace/model/AzureCloudContextFields.java

```diff
@@ -0,0 +1,106 @@
+import com.google.common.base.Objects;
+
+public class AzureCloudContextFields {
+  private final String azureTenantId;
+  private final String azureSubscriptionId;
+  private final String azureResourceGroupId;
+
+  @JsonCreator
+  public AzureCloudContextFields(
+      @JsonProperty("azureTenantId") String azureTenantId,
+      @JsonProperty("azureSubscriptionId") String azureSubscriptionId,
+      @JsonProperty("azureResourceGroupId") String azureResourceGroupId) {
+    this.azureTenantId = azureTenantId;
+    this.azureSubscriptionId = azureSubscriptionId;
+    this.azureResourceGroupId = azureResourceGroupId;
+  }
+
+  public String getAzureTenantId() {
+    return azureTenantId;
+  }
+
+  public String getAzureSubscriptionId() {
+    return azureSubscriptionId;
+  }
+
+  public String getAzureResourceGroupId() {
+    return azureResourceGroupId;
+  }
+
+  public ApiAzureContext toApi() {
+    return new ApiAzureContext()
+        .tenantId(azureTenantId)
+        .subscriptionId(azureSubscriptionId)
+        .resourceGroupId(azureResourceGroupId);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof AzureCloudContextFields that)) return false;
+    return Objects.equal(azureTenantId, that.azureTenantId)
+        && Objects.equal(azureSubscriptionId, that.azureSubscriptionId)
+        && Objects.equal(azureResourceGroupId, that.azureResourceGroupId);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(azureTenantId, azureSubscriptionId, azureResourceGroupId);
+  }
+
+  // -- serdes for the AzureCloudContext --
+
+  public String serialize() {
+    AzureCloudContextV100 dbContext =
+        AzureCloudContextV100.from(azureTenantId, azureSubscriptionId, azureResourceGroupId);
+    return DbSerDes.toJson(dbContext);
+  }
+
+  public static AzureCloudContextFields deserialize(String dbContext) {
+    try {
+      AzureCloudContextV100 result = DbSerDes.fromJson(dbContext, AzureCloudContextV100.class);
+      if (result.version != AzureCloudContextV100.AZURE_CLOUD_CONTEXT_DB_VERSION) {
+        throw new InvalidSerializedVersionException("Invalid serialized version");
+      }
+      return new AzureCloudContextFields(
+          result.azureTenantId, result.azureSubscriptionId, result.azureResourceGroupId);
+
+    } catch (SerializationException e) {
+      throw new InvalidSerializedVersionException("Invalid serialized version", e);
+    }
+  }
+
+  @VisibleForTesting
+  public static class AzureCloudContextV100 {
+    /**
+     * Format version for serialized form of Azure cloud context - deliberately chosen to be
+     * different from the GCP cloud context version.
+     */
+    public static final long AZURE_CLOUD_CONTEXT_DB_VERSION = 100;
+
+    /** Version marker to store in the db so that we can update the format later if we need to. */
+    @JsonProperty public final long version = AZURE_CLOUD_CONTEXT_DB_VERSION;
```

Comment:
this is not used, right?

also can we also do the mask thing? confusing that this is v100

Reply:
It is used and it works, even though it looks odd. The object is initialized (set to 100) before JSON is deserialized into it, so whatever is in the JSON will clobber the version.